### ISSUE TITLE
`cabal check` rejects packages which were ok previously

### ISSUE THREAD

**Issue by hvr** [haskell/cabal#3444]
This regression was introduced by 0252a0aed79e881d4f6210d9296216aef3c301d0 / #3441

```
$ cabal get base-noprelude-4.8.2.0
Downloading base-noprelude-4.8.2.0...
Unpacking to base-noprelude-4.8.2.0/
$ cd base-noprelude-4.8.2.0/
$ cabal check
The following warnings are likely affect your build negatively:
* Library base-noprelude does not expose any modules

Hackage would reject this package.
```

```
$ cabal get network-uri-2.5.0.0
Downloading network-uri-2.5.0.0...
Unpacking to network-uri-2.5.0.0/
$ cd network-uri-2.5.0.0
$ cabal check
The following warnings are likely affect your build negatively:
* Library network-uri does not expose any modules

Hackage would reject this package.
```

**Comment by hvr** [comment]
@23Skidoo btw, to me the grammar in "The following warnings are likely affect your build negatively:" seems a bit off :-)

**Comment by 23Skidoo** [comment]
@hvr 
> grammar seems a bit off 

Fixed.

**Comment by 23Skidoo** [comment]
@headprogrammingczar I suggest making this error non-fatal and changing the check to `null exposedModules && null reexportedModules`.

**Comment by headprogrammingczar** [comment]
It looks like `PackageCheck` doesn't order its constructors from most to least severe... I had not intended for it to be fatal right off the bat.

I am adding reexportedModules, but keeping libModules as it seems to be a more complete indicator that some a library's module list has been set deliberately. I will push after my local tests are done.

**Comment by headprogrammingczar** [comment]
@hvr does this PR resovle things?

**Comment by 23Skidoo** [comment]
Fixed in e9883dced04e6a16e25752cc8513ce26325b4b4b and 675748d3c687a449ebfbf712fe106709667b75a3.
